mobile/ios/ios-device.ts

```diff
@@ -320,11 +327,130 @@ export class IOSDevice implements Mobile.IIOSDevice {
 			installationProxy.deployApplication(packageFile).wait();
 			installationProxy.closeSocket();
 		}).future<void>()();
+    }
+
+    private openSafariFrontEnd(packageName: string): void {
+        var tnsIosPackage = this.$npm.install("tns-ios").wait();
+        var safariPath = path.join(tnsIosPackage, "WebInspectorUI/Safari/Main.html");
+        this.$childProcess.exec("open -a Safari " + safariPath).wait();
+    }
+
+    private attachWhenReady(packageName: string,
+        notificationProxyClient: iOSProxyServices.NotificationProxyClient): void {
```

Comment:
Do you need to pass `notificationProxyClient` as a parameter. I think you should be able to create it constructor.